src/engine/Service.cpp

```diff
@@ -112,16 +112,17 @@ ResultTable Service::computeResult() {
   // easy-to-parse format. It might not be the best choice regarding robustness
   // and portability though. In particular, we are not sure how deterministic
   // the TSV output is with respect to the precise encoding of literals.
-  std::istringstream tsvResult =
-      getTsvFunction_(serviceUrl, boost::beast::http::verb::post, serviceQuery,
-                      "application/sparql-query", "text/tab-separated-values");
+  cppcoro::generator<std::string_view> tsvResult = getTsvFunction_(
+      serviceUrl, cancellationHandle_, boost::beast::http::verb::post,
+      serviceQuery, "application/sparql-query", "text/tab-separated-values");
 
   // The first line of the TSV result contains the variable names.
-  std::string tsvHeaderRow;
-  if (!std::getline(tsvResult, tsvHeaderRow)) {
+  auto begin = tsvResult.begin();
+  if (begin == tsvResult.end()) {
```

Comment:
As stated, you should do the `getLine` here.
Implementing a `getline` based on a `generator<string_view` is a useful abstraction that can nicely be implemented.
It then again probably becomes something like a `generator<string>`.